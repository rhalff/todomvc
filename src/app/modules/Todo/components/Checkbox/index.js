// @flow
import React from 'react'
import styles from './scss/Checkbox.scss'

type Props = {
  /**
   *  Whether the component is checked
   */
  checked?: boolean,
  /**
   * Callback triggered when state is changed
   */
  onChange?: Function
}

const Checkbox = ({ checked, onChange }: Props) => {
  return (
    <input
      className={checked ? styles.checked : styles.unchecked}
      type="checkbox"
      checked={checked}
      onChange={onChange}
    />
  )
}

export default Checkbox
