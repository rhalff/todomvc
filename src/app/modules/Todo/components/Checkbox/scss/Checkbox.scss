.checkbox {
  height: 40px;
  width: 40px;
  text-align: center;
  margin: auto 0;
  border: none;
  appearance: none;
  cursor: pointer;
}

.unchecked {
  composes: checkbox;
}

.unchecked:after {
	content: url('../assets/empty.svg');
}

.checked {
  composes: checkbox;
}

.checked:after {
  content: url('../assets/check_mark.svg');
}
