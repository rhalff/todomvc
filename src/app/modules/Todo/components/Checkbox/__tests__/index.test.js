// @flow
import React from 'react'
import Checkbox from '../'
import renderer from 'react-test-renderer'

test('Can show CheckBox', () => {
  const component = renderer.create(<Checkbox onChange={() => {}} />)

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Can show CheckBox (checked)', () => {
  const component = renderer.create(<Checkbox onChange={() => {}} checked />)

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
