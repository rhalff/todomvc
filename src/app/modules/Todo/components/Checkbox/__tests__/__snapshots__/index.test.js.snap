// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can show CheckBox (checked) 1`] = `
<input
  checked={true}
  className="checked"
  onChange={[Function]}
  type="checkbox"
/>
`;

exports[`Can show CheckBox 1`] = `
<input
  checked={undefined}
  className="unchecked"
  onChange={[Function]}
  type="checkbox"
/>
`;
